storage/remote/queue_manager_test.go

```diff
@@ -1170,6 +1170,78 @@ func (c *TestWriteClient) Store(_ context.Context, req []byte, attemptNos int) e
 	return nil
 }
 
+func v2WriteRequestToWriteRequest(reqV2 *writev2.Request) (*prompb.WriteRequest, error) {
+	req := &prompb.WriteRequest{
+		Timeseries: make([]prompb.TimeSeries, len(reqV2.Timeseries)),
+		// TODO handle metadata?
```

Comment:
probably we should, though we'll lose some granularity during the conversion

the `metricmetadata` only differentiates on metric family name, so either first seen or last seen set of metadata for a metric family name wins